### ISSUE TITLE
Replace XSD screenshots with code listings

### ISSUE THREAD

**Issue by t-sommer** [modelica/fmi-standard#569]
The screenshots of the XSD schema elements are out-of-sync with the changed schema files.
They could be replaced with XML examples that can be validated during CI.

For example

![Boolean_Schema_large](https://raw.githubusercontent.com/modelica/fmi-standard/master/docs/images/Boolean_Schema_large.png)

could be replaced with

```xml
<Boolean name="bool_in"
  valueReference="8" 
  declaredType="switchState"
  start="false"/>
```

This would also save a lot of space (compared to the current PNGs), improve the rendering on small screens and provide valid (copy & paste friendly) examples.

**Comment by chrbertsch** [comment]
I have always seen these pictures as part of the normative specification.
So how can they be replaced by an example? 
Will then the XSD pert describing e.g. the attributes of the boolean variable be included in the specification? But this is not so well readable as the generated pictures.

**Comment by t-sommer** [comment]
Technically they are normative (not in square brackets). If we keep them, someone needs to update the screenshots.

Another possibility would be to rebuild them as ASCII-art. Something like

```
+ Boolean
  + declaredType (normalizedString)
  + start (boolean)
``` 

For the complex types this would be quite an effort though...

**Comment by chrbertsch** [comment]
Regular Design Meeting: 
Karl: the Screenshots give a good overview. The screenshots must be consistent from the xsd filed.
Torsten: Right now they are inconsistent. Should they be removed
Klaus: https://stackoverflow.com/questions/32563774/automatically-generate-graphical-documentation-for-xsd-like-xmlspy-does
Torsten: Someone has to do it ....
Klaus: mark it as not up-to-date. Having the pictures in the document is nice.
Torsten: Add placeholders and add the screenshots later. 
Klaus: If we do not have screenshots, are they really important?
Karl: If you are new to it, they are nice.
Klaus: it is important, what is mandatory and what not

**Comment by chrbertsch** [comment]
FMI Design meeting:
Torsten S.: I will do this.  (not super urgent)

**Comment by nickbattle** [comment]
In support of this, I made some notes of the problems I could see. Note that in some cases the **text** following the diagrams is also yet to be updated, so it's not just a matter of the screenshots being updated:

2.2.1 The top level diagram is out of date (no BuildConfiguration, TypeDefinitions don't have SimpleType any more, no ScalarVariables any more, types and variables are Float64 etc rather than Real, ModelStructure no longer as 3 subfields, etc.)

The text immediately following the diagram does not include the new top level fields (like BuildConfiguration, Terminals, GraphicalRepresentation etc.), though these do have their own sub-sections in 2.2.2, 2.2.7/8.

The fmiModelDescription attributes diagram is wrong (no instantiationToken, still has guid, though the description that follows the diagram is correct)

2.2.4 TypeDefinitions no longer have a SimpleType subfield and the types themselves are now FloatNN, (U)IntNN etc.

The type attrbute diagrams for the various types have all changed.

2.2.5 The diagram for ClockTypes has been added, but based on the old FMI2 type definition with SimpleTypes.

2.2.9 VendorAnnotations look correct, but refers to fmi2Annotation as the XSD type.

2.2.10 ModelVariables structure is still FMI2 version, though the text that follows is correct.

The attributes of Variable are still the FMI2 version, though the table that follows is correct.

The attributes of different types under the "Aliasing of Variables" section still refer to FMI2 types (fmi2IntegerAttributes  etc).

The Annotations diagram at the end of 2.2.10 looks correct, but refers to fmi2Annotations.

2.2.11 The ModelStructure type is now different, without three subfields of Unknowns. Therefore the attribute diagrams that follow (for Unknown) are also wrong.

The table description that follows is wrong, referring to the three subfields.

**Comment by chrbertsch** [comment]
Who does have xmlspy so that we could create the pictures?
@CSchulzeTLK : have you created the new pictures for the termal xsd definitions? Do you have xmlspy?

**Comment by CSchulzeTLK** [comment]
I also like the XML Spy illustrations, they are nicer are more informative than the output of other tools I have tested.

I have a license, and can update the images. There was the remark, that we should not pollute the repo with lots of image variants during development, so the images have not been updated.

For the terminals and icons, I can update all images by script, starting from the XML Spy html export. This html export includes lots of separate images.

Unfortunately lots of the images in other sections of the standard have been edited manually to generate specific views on the xsd. So these images are not part of the html export and have to be recreated manually. Maybe we can replace the images with similar views on the elements, which are part of the html export.

**Comment by CSchulzeTLK** [comment]
This is what I am using...

[export_example_with_copy_script.zip](https://github.com/modelica/fmi-standard/files/4254297/export_example_with_copy_script.zip)

**Comment by chrbertsch** [comment]
Regular Desgin meeting: 

Klaus: We could for now replace the FMI 2.0 pictures with some dummpy pictures or strikesthrough
Claudio: one could include parts of the xsd
Christian: this is not readable
Klaus: see comments above: we mark them as outdated.
Klaus: We removed the comments from the xsd. They will be smaller.

@clagms : I will mark the pictures as outdated.  possibly I do this automatically

**Comment by t-sommer** [comment]
@pmai, can you make a PR with the solution you developed for SSP, so we can use it in our CI?

**Comment by pmai** [comment]
@t-sommer sorry for the confusion, we used OxygenXML in SSP, that’s commercial and the process was manual.

I can look into xsdvi which seems scriptable, but will have to see how well this works ...

**Comment by clagms** [comment]
Working on marking the pictures as outdated now!

**Comment by clagms** [comment]
PR Open: https://github.com/modelica/fmi-standard/pull/852

**Comment by beutlich** [comment]
> 
> 
> PR Open: #852

Is there anything to do after @t-sommer squash-merged #852 earlier today? Or can this issue be closed?

**Comment by chrbertsch** [comment]
fixed by #852

**Comment by t-sommer** [comment]
Thanks @clagms and @beutlich!
